tls/s2n_fingerprint_ja3.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "tls/extensions/s2n_extension_list.h"
+#include "tls/s2n_fingerprint.h"
+#include "utils/s2n_blob.h"
+#include "utils/s2n_safety.h"
+
+#define S2N_JA3_FIELD_DIV ','
+#define S2N_JA3_LIST_DIV  '-'
+
+/* UINT16_MAX == 65535 */
+#define S2N_UINT16_STR_MAX_SIZE 5
+
+static S2N_RESULT s2n_fingerprint_ja3_digest(struct s2n_fingerprint_hash *hash,
+        struct s2n_stuffer *out)
+{
+    if (!s2n_fingerprint_hash_supports_digest(hash)) {
+        return S2N_RESULT_OK;
+    }
```

Comment:
nit: `supports_digest()` feels like the wrong name somehow. I think it's because "supported" makes it sound like we would prefer to do a digest and aren't able to, when really `s2n_fingerprint_ja3_digest()` was called without wanting a digest. Would something like `do_digest()` or `should_perform_digest()` maybe work better?

Reply:
s2n_fingerprint_hash_is_digestable? 🙃